src/files/utils.ts

```diff
@@ -0,0 +1,72 @@
+import { UseMutateAsyncFunction } from '@tanstack/react-query';
+
+import { UploadSignedUrlInput } from './schemas';
+
+export const fetchFile = (url: string, metadata?: string[]) => async () => {
+  const fileResponse = await fetch(url);
+  if (!fileResponse.ok) {
+    throw new Error('Could not fetch the file');
+  }
+
+  const lastModifiedDateHeader = fileResponse.headers.get('Last-Modified');
+
+  return (metadata || []).reduce(
+    (file, currentMetadata) => {
+      return {
+        ...file,
+        [currentMetadata]: fileResponse.headers.get(
+          `x-amz-meta-${currentMetadata}`
+        ),
+      };
+    },
+    {
+      fileUrl: url,
+      size: fileResponse.headers.get('Content-Length') ?? undefined,
+      type: fileResponse.headers.get('Content-Type') ?? undefined,
+      lastModifiedDate: lastModifiedDateHeader
+        ? new Date(lastModifiedDateHeader)
+        : new Date(),
+    }
+  );
+};
+
+export const uploadFile =
+  (
+    getPresignedUrl: UseMutateAsyncFunction<
+      { signedUrl: string; futureFileUrl: string },
+      unknown,
+      UploadSignedUrlInput | void
+    >
+  ) =>
+  async (
+    file?: File,
```

Comment:
why the file is optional ? can we make it required ? (I don't see a use case where we call uploadFile without a file ^^)

Reply:
We have to check at one point if the file retrieved from the form is empty or not.
The choice was made to handle the 'file is undefined' case in this function but could be moved in the form, before making the call to the mutation ? 

Let me know what you think is best ?